src/components/WorkerTable/index.jsx

```diff
@@ -0,0 +1,176 @@
+import { PureComponent } from 'react';
+import { Link } from 'react-router-dom';
+import { TableCell, TableRow } from 'material-ui/Table';
+import { camelCase } from 'change-case';
+import LinkIcon from 'mdi-react/LinkIcon';
+import { withStyles } from 'material-ui/styles';
+import { ListItemText } from 'material-ui/List';
+import Typography from 'material-ui/Typography';
+import ContentCopyIcon from 'mdi-react/ContentCopyIcon';
+import InformationVariantIcon from 'mdi-react/InformationVariantIcon';
+import ButtonDrawer from '../ButtonDrawer';
+import TableCellListItem from '../TableCellListItem';
+import DateDistance from '../DateDistance';
+import DataTable from '../DataTable';
+import TaskMetadataCard from '../TaskMetadataCard';
+import StatusLabel from '../StatusLabel';
+import { worker } from '../../utils/prop-types';
+import { sort } from '../../utils/helpers';
+
+@withStyles(theme => ({
+  dateListItem: {
+    marginLeft: -theme.spacing.unit,
+    padding: theme.spacing.unit,
+  },
+  listItemButton: {
+    '& svg': {
+      transition: theme.transitions.create('fill'),
+      fill: theme.palette.primary.light,
+    },
+    '&:hover svg': {
+      fill: theme.palette.common.white,
+    },
+  },
+  taskName: {
+    marginRight: theme.spacing.unit,
+    maxWidth: 250,
+    whiteSpace: 'nowrap',
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    verticalAlign: 'middle',
+    display: 'inline-block',
+  },
+  infoButton: {
+    marginLeft: -theme.spacing.double,
+    marginRight: theme.spacing.unit,
+  },
+}))
+/**
+ * Display relevant information about a worker in a table.
+ */
+export default class WorkerTable extends PureComponent {
+  static propTypes = {
+    /** A GraphQL worker response. */
+    worker,
+  };
+
+  state = {
+    sortBy: null,
+    sortDirection: null,
+  };
+
+  getTableData() {
```

Comment:
Might be interesting. Btw, `memoize` is deprecated in ramda. I'll look into using `memoizeWith`.

Reply:
When I initially tested it with `memoize`, I noticed that it became slower. I then looked at the implementation and saw that they `toString` the arguments passed to generate the cache key https://github.com/ramda/ramda/blob/master/source/memoize.js#L34. This might explain why it got deprecated :) `memoizeWith` seems to be working fine.